src/Microsoft.AspNetCore.Blazor.Build/targets/Blazor.MonoRuntime.targets

```diff
@@ -0,0 +1,305 @@
+      SkipUnchangedFiles="$(SkipCopyUnchangedFiles)"
+      OverwriteReadOnlyFiles="$(OverwriteReadOnlyFiles)"
+      Retries="$(CopyRetryCount)"
+      RetryDelayMilliseconds="$(CopyRetryDelayMilliseconds)"
+      UseHardlinksIfPossible="$(CreateHardLinksForCopyFilesToOutputDirectoryIfPossible)"
+      UseSymboliclinksIfPossible="$(CreateSymbolicLinksForCopyFilesToOutputDirectoryIfPossible)"
+      Condition="'@(BlazorItemSpec)' != '' and '$(CopyBuildOutputToOutputDirectory)' == 'true' and '$(SkipCopyBuildProduct)' != 'true'">    
+    </Copy>
+
+    <ItemGroup>
+      <BlazorOutputs Include="$(ProjectDir)$(OutputPath)dist/**/*" />
+    </ItemGroup>
+    
+  </Target>
+  
+  <Target Name="_BlazorBuildReport"
+          AfterTargets="_BlazorCopyFilesToOutputDirectory">
+
+    <PropertyGroup>
+      <BlazorBuildStatistics Condition="'$(BlazorBuildStatistics)' == ''">normal</BlazorBuildStatistics>
+      <BlazorBuildStatistics Condition="'$(BlazorBuildStatistics)' == 'true'">high</BlazorBuildStatistics>
+    </PropertyGroup>
+
+    <Message Importance="high" Text="Blazor Build result -> @(BlazorOutputs->Distinct()->Count()) files written to $(ProjectDir)$(OutputPath)dist" />
+    <Message Importance="$(BlazorBuildStatistics)" Text="%(BlazorOutputs.Identity)" />
+    </Target>
+
+  <PropertyGroup>
+    <BlazorBuildDependsOn>
+      GetBlazorBuildConfiguration;
+      ResolveBinaryOutput;
+      BuildBlazorIndexHtml;
+    </BlazorBuildDependsOn>
+  </PropertyGroup>
+
+  <Target Name="BlazorBuild" DependsOnTargets="$(BlazorBuildDependsOn)" AfterTargets="Compile" />
+
+  <Target Name="GetBlazorBuildConfiguration">
+    <!-- 
+    This task produces all the "final" paths for all the files we need to produce the final output.
+    BlazorIntermediateOutputPath is something like /obj/debug/netstandard2.0/blazor.
+    Inside there we want to produce a folder structure like the following
+    dist/_framework/_bin <- This will contain either the BCL + app assemblies or the result of linking the app.
+    dist/_framework/asmjs <- This will contain the asmjs runtime copied from the nuget package.
+    dist/_framework/wasm <- This will contain the wsm runtime copied from the nuget package.
+    dist/_framework/blazor.js <- This is the blazor.js file copied from the nuget package.
+    dist/Index.html <- This is the optional Index.html file generated from wwwroot/Index.html in case it's present.
+    
+    This task also defines some intermediate paths that we will use:
+    /obj/debug/netstandard2.0/blazor/blazor/linker <- This will be used to create the output from the linker.
+    /obj/debug/netstandard2.0/blazor/blazor/linker.marker.txt <- This will be used to save the output files from
+    the linker and use that as marker to identify whether or not we need to run the linker.
+    -->
+
+    <PropertyGroup Label="Build properties">
+      <BlazorShouldRunMonoLinker>$(BlazorLinkOnBuild)</BlazorShouldRunMonoLinker>
```

Comment:
Is there ever a scenario where BlazorShouldRunMonoLinker != BlazorLinkOnBuild? In other words, why have a separate property?

Reply:
Publish